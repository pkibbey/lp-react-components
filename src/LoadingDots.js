import React from 'react';
import { Box } from 'rebass';
import { keyframes } from '@emotion/core';

const bounce = keyframes`
  from, 10%, 27%, 40%, to {
    transform: translate3d(0,0,0);
  }
  20%, 22% {
    transform: translate3d(0, -8px, 0);
  }
  35% {
    transform: translate3d(0, -4px, 0);
  }
  45% {
    transform: translate3d(0,-1px,0);
  }
`;

const LoadingDot = ({ delay }) => (
  <Box
    as="span"
    sx={{
      backgroundColor: 'white',
      width: '3px',
      height: '3px',
      borderRadius: '50%',
      animationName: bounce,
      animationDuration: '1.5s',
      animationTimingFunction: 'linear',
      animationIterationCount: '3',
      animationDelay: `${delay}s`,
    }}
  />
);

const LoadingDots = () => (
  <Box
    as="span"
    style={{
      display: 'inline-flex',
      alignItems: 'center',
      justifyContent: 'space-between',
      boxSizing: 'border-box',
      width: '16px',
      height: '24px',
      marginLeft: '4px',
    }}
  >
    <LoadingDot delay={0} />
    <LoadingDot delay={0.15} />
    <LoadingDot delay={0.3} />
  </Box>
);

export default LoadingDots;
