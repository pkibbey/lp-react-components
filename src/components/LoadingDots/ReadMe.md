Default LoadingDots:

```js
<LoadingDots variant="dark" />
```
