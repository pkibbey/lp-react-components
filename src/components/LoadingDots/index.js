import React from 'react'
import { Box } from 'theme-ui'
import { keyframes } from '@emotion/react'
import PropTypes from 'prop-types'
import theme from '../../theme'

const bounce = keyframes`
  from, 10%, 27%, 40%, to {
    transform: translate3d(0,0,0);
  }
  20%, 22% {
    transform: translate3d(0, -8px, 0);
  }
  35% {
    transform: translate3d(0, -4px, 0);
  }
  45% {
    transform: translate3d(0,-1px,0);
  }
`

const LoadingDot = ({ delay, variant }) => (
  <Box
    as='span'
    sx={{
      backgroundColor: variant === 'light' ? 'white' : theme.colors.darkerGray,
      width: '3px',
      height: '3px',
      borderRadius: '50%',
      animation: `${bounce} 1.5s linear ${delay}s infinite`
    }}
  />
)

/**
 * Used to convey that another user or bot, is typing a message
 *
 * @visibleName Loading Dots
 */
const LoadingDots = ({ variant }) => (
  <Box
    as='span'
    sx={{
      display: 'inline-flex',
      alignItems: 'center',
      justifyContent: 'space-between',
      boxSizing: 'border-box',
      width: '17px',
      height: '24px'
    }}
  >
    <LoadingDot delay={0} variant={variant} />
    <LoadingDot delay={0.15} variant={variant} />
    <LoadingDot delay={0.3} variant={variant} />
  </Box>
)

LoadingDots.propTypes = {
  /** A variant for the loading dots */
  variant: PropTypes.oneOf(['light', 'dark'])
}

LoadingDots.defaultProps = {
  variant: 'light'
}

export default LoadingDots
