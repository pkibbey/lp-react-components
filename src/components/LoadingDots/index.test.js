import React from 'react'
import renderer from 'react-test-renderer'
import LoadingDots from './'

it('renders default loading dots', () => {
  const tree = renderer.create(<LoadingDots />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders dark loading dots', () => {
  const tree = renderer.create(<LoadingDots variant='dark' />).toJSON()
  expect(tree).toMatchSnapshot()
})
