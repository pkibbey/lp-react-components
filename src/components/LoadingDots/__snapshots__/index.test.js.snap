// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark loading dots 1`] = `
<span
  className="css-vurnku"
  style={
    Object {
      "alignItems": "center",
      "boxSizing": "border-box",
      "display": "inline-flex",
      "height": "24px",
      "justifyContent": "space-between",
      "marginLeft": "4px",
      "width": "17px",
    }
  }
>
  <span
    className="css-1m25c1c"
  />
  <span
    className="css-1q9grkm"
  />
  <span
    className="css-r0ir4z"
  />
</span>
`;

exports[`renders default loading dots 1`] = `
<span
  className="css-vurnku"
  style={
    Object {
      "alignItems": "center",
      "boxSizing": "border-box",
      "display": "inline-flex",
      "height": "24px",
      "justifyContent": "space-between",
      "marginLeft": "4px",
      "width": "17px",
    }
  }
>
  <span
    className="css-19s72f7"
  />
  <span
    className="css-oj3lh1"
  />
  <span
    className="css-t6ffo3"
  />
</span>
`;
